CONFIGURATION.md

```diff
@@ -86,14 +94,22 @@ The other placeholders are specified separately.
 
   # The IP protocol of the HTTP probe (ip4, ip6).
   [ preferred_ip_protocol: <string> | default = "ip6" ]
-  [ ip_protocol_fallback: <boolean | default = true> ]
+  [ ip_protocol_fallback: <boolean> | default = true ]
 
   # The body of the HTTP request used in probe.
   body: [ <string> ]
 
 
 ```
 
+#### <http_header_match_spec>
+
+```yml
+header: <string>,
+regexp: <regex>,
```

Comment:
This is optional

Reply:
Why is it optional? It is not possible to match a header without a regex to match against.

Reply:
Unless you allow it to be missing

Reply:
But if you allow it to be missing, then what is the point of having a match rule at all? 

```yml
fail_if_header_matches:
  - header: Transfer-Encoding
     allow_missing: true
fail_if_header_not_matches:
  - header: Transfer-Encoding
     allow_missing: true
```

Neither of these rules make sense to me. But looking at the checks made in `config.go`, apparently they used to. So I think I should change the config check, so it always required `regexp` to be set, regardless of `allow_missing`.

Would that be OK?

Reply:
In a stroke of speculative execution, did just that.